### PR TITLE
fix(engine): app_extra_logger debug level

### PR DESCRIPTION
**Pull request by pcanto-hopeit**
fix https://github.com/hopeit-git/hopeit.engine/issues/114

### CODE REVIEW

File: engine/src/hopeit/server/logger.py [hunk 1]
```diff
@@ -38,7 +38,8 @@ class EventLoggerWrapper:
     def __init__(self, logger: logging.Logger):
         self.logger = logger
 
-    def debug(self, msg, *args, **kwargs) -> None:
+    def debug(self, context: EventContext, msg, *args, **kwargs) -> None:
+        _enrich_extra(kwargs, context, msg)
```
Comment (leosmerling-hopeit) [inline engine/src/hopeit/server/logger.py:42]: We don't want to even do this call when loglevel > DEBUG, that's why enriching extra arguments is not supported in debug logging.

Comment (leosmerling-hopeit) [reply engine/src/hopeit/server/logger.py:42]: Maybe checking current log level if it does not affect performance?

Comment (leosmerling-hopeit) [reply engine/src/hopeit/server/logger.py:42]: Or capture the exception and throw a proper message indicating context/extra arguments are not supported.

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]

File: engine/src/hopeit/server/logger.py (hunk 1 above)
Comment (pcanto-hopeit) [reply engine/src/hopeit/server/logger.py:42]: Removing the context avoids any check

Comment (pcanto-hopeit) [reply engine/src/hopeit/server/logger.py]: Debug logger processor only in debug level

**Review by leosmerling-hopeit — APPROVED** [review]

**Pull request merged**